.github/workflows/build.yml

```diff
@@ -83,29 +83,34 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
+        pkg:
+          - ./client
+          - ./cmd/buildctl
+          - ./worker/containerd
+          - ./solver
```

Comment:
Some of these are really tiny. Solver has 1 integration test, containerd/worker 2,  so it really doesn't make sense to run 6 jobs for each. Only client and dockerfile are really big and need to be separate. Could we combine rest of the packages.

Reply:
Ok I've tried with the following matrix for the `workers` (new 13 jobs instead of 37):

```yaml
    matrix:
        pkg:
          - ./client
          - ./cmd/buildctl
          - ./worker/containerd
          - ./solver
          - ./frontend
          - ./frontend/dockerfile
        worker:
          - containerd*
          - oci*
        include:
          - pkg: ./client
            typ: integration
          - pkg: ./cmd/buildctl
            typ: integration
          - pkg: ./worker/containerd
            typ: integration
          - pkg: ./solver
            typ: integration
          - pkg: ./frontend/dockerfile
            typ: dockerfile
          - pkg: ''
            skip-integration-tests: 1
            typ: integration gateway
```

But as expected it [takes more time](https://github.com/moby/buildkit/actions/runs/1011821077) to complete:

![image](https://user-images.githubusercontent.com/1951866/124936038-7b6b7e80-e006-11eb-8513-1d854f2bf7c9.png)

![image](https://user-images.githubusercontent.com/1951866/124935308-d51f7900-e005-11eb-8ba2-eae463641c43.png)

~30m

Reply:
You need to combine these packages, not workers. Also, unit tests for these packages run twice. Not a big issue as all of them are fast.

Reply:
Done. 37 > 19 jobs with the same amount of time.